otel/otel.go

```diff
@@ -0,0 +1,171 @@
+package otel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	otelsdk "go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// Config defines the Otel Wrapper configuration for metrics
+type Config struct {
+	ServiceName               string
+	Debug                     bool
+	ExporterType              string
+	ExporterEndpoint          string
+	ExporterTimeout           string
+	ExporterCertificate       string
+	ExporterClientCertificate string
+	ExporterClientKey         string
+	CollectionInterval        time.Duration
+	MetricsReader             sdkmetric.Reader
+	MetricsExporter           sdkmetric.Exporter
+}
+
+// OtelWrapper encapsulates OpenTelemetry MetricProvider and utilities
+type OtelWrapper struct {
+	meterProvider *sdkmetric.MeterProvider
+	globalMeter   metric.Meter
+	config        Config
+}
+
+var (
+	globalWrapper *OtelWrapper
```

Comment:
I'm not fan of this name. We should need to find a better name.
I think the `Wrapper` term should be completely omitted in this Go package